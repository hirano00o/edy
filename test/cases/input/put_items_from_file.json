[
  {
    "ID": 15,
    "Name": "Mallet",
    "Email": "mallet@example.com",
    "Age": 20,
    "Birthday": {
      "Year": 2001,
      "Month": 8,
      "Day": 31
    }
  },
  {
    "ID": 16,
    "Name": "Marvin",
    "Email": "marvin@example.com",
    "Age": 21,
    "Birthday": {
      "Year": 2002,
      "Month": 8,
      "Day": 20
    }
  }
]
