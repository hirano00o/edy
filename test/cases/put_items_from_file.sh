#!/bin/bash

SCRIPT_ROOT_DIR=$1
TEST_NAME=$(basename "$0" | sed "s/\..*//")

# aws dynamodb batch-write-item \
#   --request-items file://${SCRIPT_ROOT_DIR}/cases/input/put_items_from_file.json \
#   --endpoint-url http://localhost:8000
CMD="edy p -t User -I ${SCRIPT_ROOT_DIR}/cases/input/put_items_from_file.json --local 8000"

. "${SCRIPT_ROOT_DIR}"/helper.sh

FILTER_CONDITION="ID,N = 15 or ID,N = 16"

run_such_put_helper
