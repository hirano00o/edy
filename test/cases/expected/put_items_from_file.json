[
  {
    "Age": 20,
    "Birthday": {
      "Day": 31,
      "Month": 8,
      "Year": 2001
    },
    "Email": "mallet@example.com",
    "ID": 15,
    "Name": "Mallet"
  },
  {
    "Age": 21,
    "Birthday": {
      "Day": 20,
      "Month": 8,
      "Year": 2002
    },
    "Email": "marvin@example.com",
    "ID": 16,
    "Name": "Marvin"
  }
]
